pkg/loop/internal/chain_reader_test.go

```diff
@@ -59,37 +60,40 @@ func TestVersionedBytesFunctions(t *testing.T) {
 }
 
 func TestChainReaderClient(t *testing.T) {
-	RunChainReaderInterfaceTests(t, &chainReaderInterfaceTester{chainReader: &fakeChainReader{lock: &sync.Mutex{}}})
+	fake := &fakeChainReader{lock: &sync.Mutex{}}
```

Comment:
nit/ if you embed the plain `sync.Mutex` type instead of a pointer, then you can just omit it and use the zero value.
```suggestion
	fake := &fakeChainReader{}
```